platforms/software/resources/src/main/java/org/gradle/internal/resource/ExternalResourceName.java

```diff
@@ -94,9 +94,9 @@ public URI getUri() {
      */
     public String getDecoded() {
         if (encodedRoot == null) {
-            return path;
+            return encode(path, false);
```

Comment:
Can you explain why this change was made?
It seems very odd that `getDecoded` returns the result of `encode`

Reply:
I think "decoded" was a bad name for this method, as it was already half-encoded (the root). I changed the name to match its usage.